### PR TITLE
Make Astrometry faster & avoid long double operations where possible

### PR DESCRIPTION
**Pull request by abhisrkckl**
Avoid creating `SkyCoord` objects during Astrometry computations. This was partly implemented in #1646 for models with proper motion. This PR does the same thing for cases without proper motion. This has a significant performance advantage.

Also, avoid long double operations where possible. This is done by including a new column `tdbfloat` in the TOAs table. This is the same thing as `tdbld`, but reduced to float64 precision.

Includes some refactorings suggested by sourcery.

### CODE REVIEW

**Comment by abhisrkckl** [comment]
**Profiling test**
(using the NGC6440E example dataset.)

Before
----------
image

After
------
image

i.e., a 45% reduction in execution time.

**Comment by dlakaplan** [comment]
Have you verified that going to float instead of long double doesn't change precision?

File: src/pint/models/pulsar_binary.py [hunk 1]
```diff
@@ -377,7 +376,7 @@ def update_binary_object(self, toas, acc_delay=None):
                 # it's already in ICRS
                 updates["obs_pos"] = tbl["ssb_obs_pos"].quantity
                 updates["psr_pos"] = self._parent.ssb_to_psb_xyz_ICRS(
-                    epoch=tbl["tdbld"].astype(np.float64)
+                    epoch=tbl["tdbld"]
```
Comment (dlakaplan) [inline src/pint/models/pulsar_binary.py:379]: Should this be the ld version, or the float version?

File: src/pint/models/pulsar_binary.py [hunk 2]
```diff
@@ -390,7 +389,7 @@ def update_binary_object(self, toas, acc_delay=None):
                     PulsarEcliptic(ecl=self._parent.ECL.value)
                 ).cartesian.xyz.transpose()
                 updates["psr_pos"] = self._parent.ssb_to_psb_xyz_ECL(
-                    epoch=tbl["tdbld"].astype(np.float64), ecl=self._parent.ECL.value
+                    epoch=tbl["tdbld"], ecl=self._parent.ECL.value
```
Comment (dlakaplan) [inline src/pint/models/pulsar_binary.py:392]: Should this be the ld version, or the float version?

Comment (abhisrkckl) [reply src/pint/models/pulsar_binary.py]: changed.

File: src/pint/models/pulsar_binary.py (hunk 1 above)
Comment (abhisrkckl) [reply src/pint/models/pulsar_binary.py]: changed.

**Comment by abhisrkckl** [comment]
Closing this because the effects of the two separate changes are difficult to test together. 
I'll make separate PRs for each.

**Pull request closed without merging**